cpp/examples/nested_types/deduplication.cpp

```diff
@@ -0,0 +1,205 @@
+  std::vector<cudf::groupby::aggregation_request> requests;
+  requests.emplace_back(cudf::groupby::aggregation_request());
+  auto agg = cudf::make_count_aggregation<cudf::groupby_aggregation>();
+  requests[0].aggregations.push_back(std::move(agg));
+  requests[0].values = *val;
+  auto agg_results   = grpby_obj.aggregate(requests);
+  auto result_key    = std::move(agg_results.first);
+  auto result_val    = std::move(agg_results.second[0].results[0]);
+
+  auto left_cols = result_key->release();
+  left_cols.push_back(std::move(result_val));
+
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+/**
+ * @brief Join each row with its duplicate counts
+ */
+std::unique_ptr<cudf::table> join_count(cudf::table_view left, cudf::table_view right)
+{
+  auto [left_indices, right_indices] =
+    cudf::inner_join(cudf::table_view{{left.column(0)}}, cudf::table_view{{right.column(0)}});
+  auto new_left  = cudf::gather(left, cudf::device_span<int const>{*left_indices});
+  auto new_right = cudf::gather(right, cudf::device_span<int const>{*right_indices});
+
+  auto left_cols  = new_left->release();
+  auto right_cols = new_right->release();
+  left_cols.push_back(std::move(right_cols[1]));
+
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+/**
+ * @brief Sort nested-type column
+ */
+std::unique_ptr<cudf::table> sort_keys(cudf::table_view tbl)
+{
+  auto sort_order = cudf::sorted_order(cudf::table_view{{tbl.column(0)}});
+  return cudf::gather(tbl, *sort_order);
+}
+
+/**
+ * @brief Main for nested_types examples
+ *
+ * Command line parameters:
+ * 1. JSON input file name/path (default: "example.json")
+ * 2. JSON output file name/path (default: "output.json")
+ * 3. Memory resource (optional): "pool" or "cuda" (default: "pool")
+ *
+ * The stdout includes the number of rows in the input and the output size in bytes.
+ */
+int main(int argc, char const** argv)
+{
+  std::string input_filepath;
+  std::string output_filepath;
+  std::string mr_name;
+  if (argc < 4) {
+    input_filepath  = "example.json";
+    output_filepath = "output.json";
+    mr_name         = "pool";
+  } else if (argc == 4) {
+    input_filepath  = argv[1];
+    output_filepath = argv[2];
+    mr_name         = argv[3];
+  } else {
+    std::cout << "Either provide all command-line arguments, or none to use defaults" << std::endl;
+    return 1;
+  }
+
+  auto resource = create_memory_resource(mr_name);
+  rmm::mr::set_current_device_resource(resource.get());
+
+  // read input file
+  auto [tbl, metadata] = read_json(input_filepath);
+
+  auto const st = std::chrono::steady_clock::now();
+
+  auto const count_st                            = std::chrono::steady_clock::now();
+  auto count                                     = count_aggregate(tbl->view());
+  std::chrono::duration<double> const count_time = std::chrono::steady_clock::now() - count_st;
+  std::cout << "count_aggregate time: " << count_time.count() << " seconds\n";
```

Comment:
Is showing timing results helpful when the example input is so small?

Reply:
The example input is just an example, after all. You could use your own JSON input file with this.

Reply:
I agree, but I think it's unlikely that a user will want to use this exact pipeline. In other words, it's not helpful to show timing if the provided input example is so small. I think you have two choices:
1. just execute the pipeline without showing timing, since you're mostly just measuring launch latency.
2. write a script that generates a sufficiently large input file (where the timing becomes a meaningful measure of speed) and ship it with this example.

Reply:
@GregoryKimball should we ship your script that generates large inputs?

Reply:
thought: I'm inclined to say we should drop the timings. The performance of this example is not itself going to be valuable.